packages/xod-project/src/patch.js

```diff
@@ -1237,3 +1238,53 @@ export const validatePatchForVariadics = def(
     Either.of
   )(patch)
 );
+
+/**
+ * Computes and returns map of pins for Node with additional Pins,
+ * if Node has `arityLevel > 1` and Patch has a variadic markers.
+ */
+export const addVariadicPins = def(
+  'addVariadicPins :: Node -> Patch -> Map PinKey Pin -> Map PinKey Pin',
+  (node, patch, originalPins) => {
+    const arityLevel = Node.getNodeArityLevel(node);
+    const isVariadic = isVariadicPatch(patch);
+    if (arityLevel === 1 || !isVariadic) return originalPins;
+
+    const variadicPins = listVariadicValuePins(patch);
+    if (Either.isLeft(variadicPins)) return originalPins;
+
+    const pinsToRepeat = explodeEither(variadicPins);
+    const variadicCount = pinsToRepeat.length;
+    const newPins = R.times(
+      idx => R.map(
+        originalPin => R.compose(
+          newPin => Pin.setPinValue(
+            Node.getBoundValueOrDefault(newPin, node),
+            newPin
+          ),
+          () => Pin.createPin(
+            Pin.addVariadicPinKeySuffix(
+              (idx + 1),
+              Pin.getPinKey(originalPin)
+            ),
+            Pin.getPinType(originalPin),
+            Pin.getPinDirection(originalPin),
+            Pin.getPinOrder(originalPin) + (variadicCount * (idx + 1)),
+            Pin.induceVariadicPinLabel(idx, Pin.getPinLabel(originalPin)),
```

Comment:
Why not add more "private" setters in addition to `setPinLabel` and `setPinValue`?